internal/util/otelfeaturegatefix/featuregate_override.go

```diff
@@ -0,0 +1,13 @@
+package otelfeaturegatefix
+
+import "go.opentelemetry.io/collector/featuregate"
+
+func init() {
+	// Override the default behavior of the feature gate to not panic when a gate is already registered.
+	// TODO: Remove this once https://github.com/prometheus/prometheus/issues/13842 is completed and we upgraded Prometheus.
+	featuregate.GlobalRegistry().SetAlreadyRegisteredErrHandler(
```

Comment:
Does this need to be an init or can we move it to normal execution, for example in entrypoint?

Reply:
There is more than one code path where this is needed -  mostly because of tests we inherited from the static mode etc. I chose to use an `init()` function to make adding this to your package as simple as adding one-line import. But using `init()` is not mandatory. I just think it's more convenient.

Reply:
Make the init for tests a `_test.go` and the init for entrypoint specific. I bring it up because init order can change and if for some reason we import a library that overrides the handler then we might get bad behavior. If not its not a huge deal, just dealt with a lot of inits in our upstream packages and dislike them.

Reply:
I tried this and unfortunately it's been taking me way too long time to figure out how to do it differently. We can maybe do this in a separate PR if you feel strong about it. There are import paths that result in libraries init code trying to register a feature gate before we can add the override handler and it's hard to find them all and the best insertion point. I think what we have now is at least doing the job and the good news is that with the next upgrade of Prometheus we should be able to get rid of this fork altogether.